packages/io/svg-deserializer/index.js

```diff
@@ -220,50 +231,51 @@ const codify = function (group) {
       svgUnitsX,
       svgUnitsY,
       svgUnitsV,
-      svgGroups
+      svgGroups,
+      target
     }
 
     let tmpCode = shapesMapJscad(obj, codify, params)
     code += tmpCode
 
-    if ('fill' in obj) {
-    // FIXME when CAG supports color
-    //  code += indent+on+' = '+on+'.setColor(['+obj.fill[0]+','+obj.fill[1]+','+obj.fill[2]+']);\n';
-    }
     if ('transforms' in obj) {
       // NOTE: SVG specifications require that transforms are applied in the order given.
       //       But these are applied in the order as required by CSG/CAG
-      let tr = null;
-      let ts = null;
-      let tt = null;
-
+      let tr
+      let ts
+      let tt
       for (let j = 0; j < obj.transforms.length; j++) {
-        var t = obj.transforms[j]
+        let t = obj.transforms[j]
         if ('rotate' in t) { tr = t }
         if ('scale' in t) { ts = t }
         if ('translate' in t) { tt = t }
       }
-      if (ts !== null) {
+      if (ts) {
         const x = ts.scale[0]
         const y = ts.scale[1]
-        code += indent + on + ' = ' + on + '.scale([' + x + ',' + y + ']);\n'
+        code += `${indent}${on} = transforms.scale([${x}, ${y}, 1], ${on})\n`
       }
-      if (tr !== null) {
-        console.log('tr', tr)
-        const z = 0 - tr.rotate
-        code += indent + on + ' = ' + on + '.rotateZ(' + z + ');\n'
+      if (tr) {
+        const z = 0 - tr.rotate * 0.017453292519943295 // radians
+        code += `${indent}${on} = transforms.rotateZ(${z}, ${on})\n`
       }
-      if (tt !== null) {
+      if (tt) {
         const x = cagLengthX(tt.translate[0], svgUnitsPmm, svgUnitsX)
         const y = (0 - cagLengthY(tt.translate[1], svgUnitsPmm, svgUnitsY))
-        code += indent + on + ' = ' + on + '.translate([' + x + ',' + y + ']);\n'
+        code += `${indent}${on} = transforms.translate([${x}, ${y}, 0], ${on})\n`
       }
     }
-    code += indent + ln + ' = ' + ln + '.union(' + on + ');\n'
+    if (target === '1D' && obj.stroke) {
```

Comment:
I do not quite understand how this works, could you explain it a bit please? thanks :)

Reply:
Basically, SVG stroke is used for color() on path2 objects, which SVG fill is used for color() on geom2 objects.

Reply:
ok great!